.github/workflows/pre-commit.yml

```diff
@@ -9,6 +9,6 @@ jobs:
   pre-commit:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
-    - uses: actions/setup-python@v2
-    - uses: pre-commit/action@v2.0.3
+    - uses: actions/checkout@v4
```

Comment:
Let's directly go to `actions/checkout@v5`. Similar in `.github/workflows/check.yml`.